### PR TITLE
add load-resnet from caffemodel

### PR DESCRIPTION
**Pull request by erichuang0771**
Add "load-resnet" example that is able to load resnet-50, 101 and 152 from caffemodels.

The model is implemented based on the Tensorpack resnet model, however some modifications are added to match the orginal Caffe implentation in Kaiming's repo.

### CODE REVIEW

File: examples/ResNet/load-resnet.py [hunk 1]
```diff
@@ -0,0 +1,229 @@
...
+            l = Conv2D('conv2', l, ch_out, 3, stride=stride)
+            l = BatchNorm('bn2', l)
+            l = tf.nn.relu(l)
+            l = Conv2D('conv3', l, ch_out * 4, 1)
+            l = BatchNorm('bn3', l)  # put bn at the bottom
+            return l + caffe_shortcut(input, ch_in, ch_out * 4, stride)
+
+        def layer(l, layername, block_func, features, count, stride, first=False):
+            if count < 1:
+                return l
```
Comment (ppwwyyxx) [inline examples/ResNet/load-resnet.py]: seems that this is not used (my bad)

File: examples/ResNet/load-resnet.py [hunk 2]
```diff
@@ -0,0 +1,229 @@
...
+        session_init=ParamRestore(resNet_param),
+        session_config=get_default_sess_config(0.9),
+        output_var_names=['prob_output']   # output:0 is the probability distribution
+    )
+    predict_func = get_predict_func(pred_config)
+
+    import cv2
+    import csv
+    remap_func = lambda  x: (x  - image_mean * 255)
+    # im = cv2.imread( '/home/eric/re-id/train/tensorpack/cat.jpg' )
```
Comment (ppwwyyxx) [inline examples/ResNet/load-resnet.py]: please clean the unused debug code

File: examples/ResNet/load-resnet.py [hunk 3]
```diff
@@ -0,0 +1,229 @@
...
+import scipy.io as sio
+import random
+import csv
+from tensorflow.contrib.layers import variance_scaling_initializer
+
+from tensorpack import *
+from tensorpack.tfutils.symbolic_functions import *
+from tensorpack.tfutils.summary import *
+
+from DataLayer import  ReIDDataLayer
```
Comment (erichuang0771) [inline examples/ResNet/load-resnet.py]: nvm.... delete please....

File: examples/ResNet/load-resnet.py [hunk 4]
```diff
@@ -0,0 +1,229 @@
...
+from tensorpack.dataflow.dataset import ILSVRCMeta
+
+"""
+Usage:
+    python2 -m tensorpack.utils.loadcaffe PATH/TO/CAFFE/{ResNet-101-deploy.prototxt,ResNet-101-model.caffemodel} ResNet101.npy
+    ./load-alexnet.py --load ResNet-101.npy --input cat.png
```
Comment (erichuang0771) [inline examples/ResNet/load-resnet.py]: yes... you need..

**Comment by erichuang0771** [comment]
Done! All requested modifications are added and also tested : )

**Comment by ppwwyyxx** [comment]
LGTM. Thanks!

**Pull request merged**